addons/httprox/common/src/main/java/org/commonjava/indy/httprox/handler/ProxyResponseWriter.java

```diff
@@ -367,62 +351,181 @@ private EventMetadata createEventMetadata( final boolean writeBody, final UserPa
         return eventMetadata;
     }
 
-    private RemoteRepository getRepository( final URL url )
-                    throws IndyDataException
+    private TrackingKey getTrackingKey( UserPass proxyUserPass ) throws IndyWorkflowException
     {
-        int port = url.getPort();
-        if ( port < 1 )
+        TrackingKey tk = null;
+        switch ( config.getTrackingType() )
         {
-            port = url.getDefaultPort();
-        }
+            case ALWAYS:
+            {
+                if ( proxyUserPass == null )
+                {
+                    throw new IndyWorkflowException( ApplicationStatus.BAD_REQUEST.code(),
+                                                     "Tracking is always-on, but no username was provided! Cannot initialize tracking key." );
+                }
 
-        String name = PROXY_REPO_PREFIX + url.getHost().replace( '.', '-' ) + '_' + port;
+                tk = new TrackingKey( proxyUserPass.getUser() );
 
-        final String baseUrl = String.format( "%s://%s:%s/", url.getProtocol(), url.getHost(), port );
+                break;
+            }
+            case SUFFIX:
+            {
+                if ( proxyUserPass != null )
+                {
+                    final String user = proxyUserPass.getUser();
 
-        ArtifactStoreQuery<RemoteRepository> query =
-                storeManager.query().packageType( GENERIC_PKG_KEY ).storeType( RemoteRepository.class );
+                    // TODO: Will this always be non-null here? Can we have an unsecured proxy?
+                    if ( user.endsWith( TRACKED_USER_SUFFIX ) && user.length() > TRACKED_USER_SUFFIX.length() )
+                    {
+                        tk = new TrackingKey( StringUtils.substring( user, 0, - TRACKED_USER_SUFFIX.length() ) );
+                    }
+                }
 
-        RemoteRepository remote = query.getRemoteRepositoryByUrl( baseUrl );
-        if ( remote == null )
-        {
-            logger.debug( "Looking for remote repo with name: {}", name );
-            remote = query.getByName( name );
-            // if repo with this name already exists, it has a different url, so we need to use a different name
-            int i = 1;
-            while ( remote != null )
+                break;
+            }
+            default:
             {
-                name = PROXY_REPO_PREFIX + url.getHost().replace( '.', '-' ) + "_" + i++;
-                logger.debug( "Looking for remote repo with name: {}", name );
-                remote = query.getByName( name );
             }
         }
+        return tk;
+    }
+
+    private RemoteRepository getRemoteRepository( String trackingId, final URL url )
```

Comment:
Could we have this return an ArtifactStore, and then just return the Group in the event a trackingId is used? The `transfer()` method would have to adjust I guess, but I think it's using the ContentManager, so it shouldn't be too difficult I think...

Reply:
Ok. So the transfer will be retrieved from the group.

Reply:
Yep